app/src/main/java/com/joedae/propertylist/presentation/MainActivity.kt

```diff
@@ -17,47 +14,54 @@ import com.joedae.propertylist.domain.GetPropertyUseCase
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
-    private var listings: MutableList<List<Property>> = mutableListOf()
 
-    val favoritesUseCase = FavoritesUseCase(favoritesRepo = FavoritesRepo(FavoritesDatabase.getInstance(PropertyComponent.getContext()).favoritesDao()))
-    var propertyViewModel = PropertyViewModel(getPropertyUseCase = GetPropertyUseCase(), favoritesUseCase)
+    val favoritesUseCase = FavoritesUseCase(
+        favoritesRepo = FavoritesRepo(
+            FavoritesDatabase.getInstance(PropertyComponent.getContext()).favoritesDao()
+        )
+    )
+
+    var propertyViewModel =
+        PropertyViewModel(getPropertyUseCase = GetPropertyUseCase(), favoritesUseCase)
 
     private val setFavorite: SetFavorite = object : SetFavorite {
         override fun onSetFavorite(id: String) {
             propertyViewModel.setFavorite(id)
         }
     }
 
+    val adapter = CustomAdapter(listOf(), setFavorite)
+
     init {
-        propertyViewModel.startListenToFavoritesChanges()
+        propertyViewModel.getListings()
+        propertyViewModel.getFavoritesUpdates()
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         binding.list.layoutManager = LinearLayoutManager(this)
-        val view = binding.root
-        setContentView(view)
+        binding.list.adapter = adapter
+        setContentView(binding.root)
 
         propertyViewModel.propertyData.observe(this) {
-            val adapter = CustomAdapter(it.results, setFavorite)
-            Toast.makeText(this, adapter.itemCount.toString(), Toast.LENGTH_SHORT).show()
-
-            binding.list.adapter = adapter
-            listings = mutableListOf(it.results)
+            adapter.properties = it.results
+            // Once we get Favorite flags we update the UI
```

Comment:
@Paycraft most of it is just code refactoring, this is the main difference:

We have a 1st Flow to get Favourites when the list is loaded, so we know to set their flags.